src/model_io/urdf/src/URDFModelExport.cpp

```diff
@@ -0,0 +1,345 @@
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include "URDFModelExport.h"
+
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/SpatialInertia.h>
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/Model/Traversal.h>
+#include <iDynTree/Model/FixedJoint.h>
+#include <iDynTree/Model/PrismaticJoint.h>
+#include <iDynTree/Model/RevoluteJoint.h>
+
+#include "URDFParsingUtils.h"
+
+
+#include <iomanip>
+#include <fstream>
+#include <string>
+
+#include <libxml/tree.h>
+#include <libxml/parser.h>
+
+namespace iDynTree
+{
+
+/*
+ * Add a <origin> URDF element to the specified parent element with the specified transform.
+ *
+ * If parent_element contains a pointer to a tag <parent_element></parent_element>, this function
+ * modifies it to be something like:
+ * <parent_element>
+ *   <origin xyz="1.0 2.0 3.0" rpy="0.0 1.0 2.0" />
+ * </parent_element>
+ * where the actual values of the xyz and rpy attributes depend on the input trans argument.
+ */
+bool exportTransform(const Transform &trans, xmlNodePtr parent_element)
+{
+    bool ok = true;
+    xmlNodePtr origin = xmlNewChild(parent_element, NULL, BAD_CAST "origin", NULL);
+    std::string pose_xyz_str;
+    ok = ok && vectorToString(trans.getPosition(), pose_xyz_str);
+    std::string pose_rpy_str;
+    Vector3 pose_rpy = trans.getRotation().asRPY();
+    ok = ok && vectorToString(pose_rpy, pose_rpy_str);
+    xmlNewProp(origin, BAD_CAST "xyz", BAD_CAST pose_xyz_str.c_str());
+    xmlNewProp(origin, BAD_CAST "rpy", BAD_CAST pose_rpy_str.c_str());
+    return ok;
+}
+
+/*
+ * Add a <inertial> URDF element to the specified parent element with the specified 6D inertia.
+ *
+ * If parent_element contains a pointer to a tag <parent_element></parent_element>, this function modifies it to be something like:
+ * <parent_element>
+ *   <inertial>
+ *     <origin xyz="0 0 0.5" rpy="0 0 0"/>
+ *     <mass value="1"/>
+ *     <inertia ixx="100"  ixy="0"  ixz="0" iyy="100" iyz="0" izz="100" />
+ *   </inertial>
+ * </parent_element>
+ * where the actual values of the origin, mass and inertia element's attributes depend on the input inertia argument.
+ */
+bool exportInertial(const SpatialInertia &inertia, xmlNodePtr parent_element)
+{
+    bool ok=true;
+    std::string bufStr;
+    xmlNodePtr inertial = xmlNewChild(parent_element, NULL, BAD_CAST "inertial", NULL);
+
+
+    xmlNodePtr mass_xml = xmlNewChild(inertial, NULL, BAD_CAST "mass", NULL);
+    ok = ok && doubleToStringWithClassicLocale(inertia.getMass(), bufStr);
+    xmlNewProp(mass_xml, BAD_CAST "value", BAD_CAST bufStr.c_str());
+
+    ok = ok && exportTransform(Transform(Rotation::Identity(), inertia.getCenterOfMass()), inertial);
+
+    xmlNodePtr inertia_xml = xmlNewChild(inertial, NULL, BAD_CAST "inertia", NULL);
+    RotationalInertiaRaw rotInertia = i.getRotationalInertiaWrtCenterOfMass();
```

Comment:
I think this one was also meant to be `inertia `and not `i`

Reply:
That happens when you modify the code and you do not try to recompile it. : ) 
Thanks for the check.